Create popup info view

iOSKirill opened the pull request:
1. Added popup info view. 
2. There is a problem when opening popup info, the blur is not superimposed on the button that is located in the NavigationBar.

iOSKirill committed:
ADDED Popup info view

iOSKirill committed:
FIXED indent bottom

vkostechko-applace reviewed (commented):
<img width="1124" alt="HeartRateMonitorApp_—_CustomPopupVStackInfo_swift" src="https://github.com/iOSKirill/Heart-Rate-Monitor-and-Pulse-App/assets/152169284/4b29cb85-313c-4ba2-a978-92ddd76a3816">
please fix warnings

vkostechko-applace reviewed (commented):
<img width="803" alt="iPhone_15_Pro_and_Heart_Rate_Monitor___Pulse_App_and_HeartRateMonitorApp_—_TabBarView_swift" src="https://github.com/iOSKirill/Heart-Rate-Monitor-and-Pulse-App/assets/152169284/344a8dfc-7d32-437e-a83b-3e3ff30be446">

vkostechko-applace reviewed (commented):
![simulator_screenshot_5CD7F093-4352-4E5C-8745-04EC614E1566](https://github.com/iOSKirill/Heart-Rate-Monitor-and-Pulse-App/assets/152169284/79f36889-a78f-4d11-a7ee-dedf8a62f9fa)

iOSKirill committed:
ADDED Font+Extension

iOSKirill committed:
ADDED Localization button

iOSKirill committed:
ADDED  Present popupInfo with animation

iOSKirill committed:
FIXED NavigationBar

Made a custom navigation Bar and fixed shadow overlay when opening popupInfo

iOSKirill commented on HeartRateMonitorApp/HeartRateMonitorApp/HomeHealthModule/View/HomeHealthView.swift:
```diff
@@ -226,12 +277,17 @@ struct HomeHealthView: View {
+                    Color.black.opacity(0.6)
+                        .edgesIgnoringSafeArea(.all)
+                    popupInfo
+                }
```
Fixed

iOSKirill commented on HeartRateMonitorApp/HeartRateMonitorApp/HomeHealthModule/View/HomeHealthView.swift:
```diff
@@ -60,6 +60,57 @@ struct HomeHealthView: View {
+                    viewModel.isPopupVisible.toggle()
+                } label: {
+                    VStack {
+                        Text("OK")
```
Fixed

iOSKirill committed:
FIXED Color TabBar buttons

The color of the active button on the tabBar has changed

iOSKirill committed:
FIXED Bugs

Fixed shadow overlay on TabBar, variable names and PopupInfoView animation appearance time

iOSKirill commented on HeartRateMonitorApp/HeartRateMonitorApp/HomeHealthModule/View/HomeHealthView.swift:
```diff
@@ -213,26 +251,36 @@ struct HomeHealthView: View {
+                Color.black.opacity(0.6)
+                    .edgesIgnoringSafeArea(.all)
+                popupInfo
+                    .statusBar(hidden: true)
```
Fixed

iOSKirill commented on HeartRateMonitorApp/HeartRateMonitorApp/HomeHealthModule/View/HomeHealthView.swift:
```diff
@@ -61,16 +45,70 @@ struct HomeHealthView: View {
                 Spacer()
                 Button {
-                    // Additional information
+                    withAnimation(.linear(duration: 0.5)) {
```
Fixed

iOSKirill commented on HeartRateMonitorApp/HeartRateMonitorApp/HomeHealthModule/View/HomeHealthView.swift:
```diff
@@ -61,16 +45,70 @@ struct HomeHealthView: View {
                 Spacer()
                 Button {
-                    // Additional information
+                    withAnimation(.linear(duration: 0.5)) {
```
Fixed

iOSKirill commented on HeartRateMonitorApp/HeartRateMonitorApp/CustomViews/CustomNavigationBar.swift:
```diff
@@ -0,0 +1,56 @@
+struct CustomNavigationBar: View {
+    // MARK: - Property -
+    @StateObject var viewModel = HomeHealthViewModel()
+    @Binding var isScroll: Bool
```
Fixed

iOSKirill commented on HeartRateMonitorApp/HeartRateMonitorApp/CustomViews/CustomNavigationBar.swift:
```diff
@@ -0,0 +1,56 @@
+
+import SwiftUI
+
+struct ScrollPreKey: PreferenceKey {
```
Fixed
